theseus/core/cost_function.py

```diff
@@ -187,11 +203,42 @@ def _compute_autograd_jacobian(
             vectorize=self._autograd_vectorize,
         )
 
-    # Returns (jacobians, error)
+    def _make_jac_fn_functorch(
+        self, tmp_optim_vars: Tuple[Manifold, ...], tmp_aux_vars: Tuple[Variable, ...]
+    ):
+        def jac_fn(optim_vars_tensors_, aux_vars_tensors_):
+            assert len(optim_vars_tensors_) == len(tmp_optim_vars)
+
+            with enable_functorch():
```

Comment:
Lots going on here, it would be useful to have some comments explaining what's happening. This is what I understand:

- Need to disable Lie Group checks that are not compatible with `functorch`, which is why we use the context.
- `functorch` passes tensors that already indexed within the batch, which means we need to add back the batch dimension with `unsqueeze(0)`.
- The error function will return a batch dimension, but `functorch` expects no batch dimension, hence we return `[0]`. 
- Unlike the other methods, we need to update the `aux_vars` every jacobian call so that the variable contains **only** the current batch element being evaluated. 

Is this correct? If so, can. you add this explanations at the top of `_make_jac_fn_functorch`?

Reply:
From Luis "you can just have a single function body for this test, like this one, and just loop over the LieGroup class and group function you want to test"